samples/tls_between_enclaves/AttestedTLSREADME.md

```diff
@@ -0,0 +1,153 @@
+
+#### Generate TLS certificate
+
+  A connecting party needs to provide a key pair for the oe_generate_attestation_certificate api to produce a self-signed certificate. These keys could be transient keys and unique for each new TLS connection.
+  - a private key (pkey): used for generating a certificate and represent the identity of the TLS connecting party
+  - a public key (pubkey): used in the TLS handshake process to create a digital signature in every TLS connection,
+
+```
+/**
+ * oe_generate_attestation_certificate.
+ *
+ * This function generates a self-signed x.509 certificate with an embedded
+ * quote from the underlying enclave.
+ *
+ * @param[in] subject_name a string contains an X.509 distinguished
+ * name (DN) for customizing the generated certificate. This name is also used
+ * as the issuer name because this is a self-signed certificate
+ * See RFC5280 (https://tools.ietf.org/html/rfc5280) specification for details
+ * Example value "CN=Open Enclave SDK,O=OESDK TLS,C=US"
+ *
+ * @param[in] private_key a private key used to sign this certificate
+ * @param[in] private_key_size The size of the private_key buffer
+ * @param[in] public_key a public key used as the certificate's subject key
+ * @param[in] public_key_size The size of the public_key buffer.
+ *
+ * @param[out] output_cert a pointer to buffer pointer
+ * @param[out] output_cert_size size of the buffer above
+ *
+ * @return OE_OK on success
+ */
+oe_result_t oe_generate_attestation_certificate(
+    const unsigned char* subject_name,
+    uint8_t* private_key,
+    size_t private_key_size,
+    uint8_t* public_key,
+    size_t public_key_size,
+    uint8_t** output_cert,
+    size_t* output_cert_size);
+```
+#### Authenticate peer certificate
+
+Upon receiving a certificate from the peer endpoint, a connecting party needs to perform peer certificate validation.
+
+In this feature, instead of using the TLS API's default authentication routine, which validates the certificate against a pre-determined CAs for authentication, an application needs to conduct "Extended custom certificate validation" inside the peer custom certificate verification callback (cert_verify_callback), which is supported by all the popular TLS APIs.
+
+```
+For example:
+
+    Mbedtls:
+            void mbedtls_ssl_conf_verify(
+                      mbedtls_ssl_config *conf,
+                      int(*f_vrfy)(void *, mbedtls_x509_crt *, int, uint32_t *)
+                      void *p_vrfy)
+
+    OpenSSL:
+            void SSL_CTX_set_verify(
+                      SSL_CTX *ctx, int mode,
+                      int (*verify_callback)(int, X509_STORE_CTX *))
+```
+##### Custom extended certificate validation
+
+The following four validation steps are performed inside the cert_verify_callback
+  1. Validate certificate
+     - Verify the signature on the self-signed certificate to ascertain that the attestation report is genuine and unmodified.
```

Comment:
Change "on the self-signed" to "of the self-signed"

Reply:
addressed